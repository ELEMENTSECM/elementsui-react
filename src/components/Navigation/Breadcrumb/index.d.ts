import * as React from 'react';

export interface BreadcrumbProps {
	/**
	 * Breadcrumb arialabel for navigation landmark
	 */
	ariaLabel?: string;
	/**
	 * Breadcrumb max visible crumbs when availible space
	 */
	maxDisplayedItems?: number;
	/**
	 * The actual crumb elements
	 */
	items?: any[];
	/**
	 * User-defined styling
	 */
	styles?: (...args: any[]) => any;
}

declare const Breadcrumb: React.SFC<BreadcrumbProps>;

export default Breadcrumb;
