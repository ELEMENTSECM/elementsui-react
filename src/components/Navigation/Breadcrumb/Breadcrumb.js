import React from 'react';
import PropTypes from 'prop-types';
import { Breadcrumb as UIFabBreadcrumb } from 'office-ui-fabric-react/lib/Breadcrumb';
import { getStyles } from './Breadcrumb.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';

export function Breadcrumb(props) {
	const { items, ariaLabel, maxDisplayedItems, className, theme, getStyles } = props;
	const classNames = classNamesFunction()(getStyles, props);
	return (
		<UIFabBreadcrumb
			className={classNames.root}
			ariaLabel={ariaLabel}
			maxDisplayedItems={maxDisplayedItems}
			items={items}
		/>
	);
}

Breadcrumb.propTypes = {
	/** Breadcrumb arialabel for navigation landmark*/
	ariaLabel: PropTypes.string,
	/** Breadcrumb max visible crumbs when availible space */
	maxDisplayedItems: PropTypes.number,
	/** The actual crumb elements */
	items: PropTypes.array,
	/** User-defined styling */
	getStyles: PropTypes.func
};

export default styled(customizable('Breadcrumb', ['theme'])(Breadcrumb), getStyles);
