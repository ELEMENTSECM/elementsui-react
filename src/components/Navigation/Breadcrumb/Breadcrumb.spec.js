import React from 'react';
import renderer from 'react-test-renderer';
import Breadcrumb from './Breadcrumb';

describe('Breadcrumb', () => {
	test('displays `Default Breadcrumb` as ariaLabel', () => {
		const tree = renderer
			.create(
				<Breadcrumb id="defaultBreadcrumb" ariaLabel="Default Breadcrumb" items={['1', '2']} />
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
