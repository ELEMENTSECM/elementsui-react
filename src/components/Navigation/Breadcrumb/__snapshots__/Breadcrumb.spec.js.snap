// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb displays \`Default Breadcrumb\` as ariaLabel 1`] = `
<div
  className="ms-ResizeGroup root-49"
>
  <div
    style={
      Object {
        "position": "fixed",
        "visibility": "hidden",
      }
    }
  >
    <div
      aria-label="Default Breadcrumb"
      className="ms-Breadcrumb root-41"
      role="navigation"
    >
      <div
        className="ms-FocusZone"
        data-focuszone-id="FocusZone1"
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDownCapture={[Function]}
        role="presentation"
      >
        <ol
          className="ms-Breadcrumb-list list-42"
        >
          <li
            className="ms-Breadcrumb-listItem listItem-43"
          >
            <span
              className="ms-Breadcrumb-item item-48"
            >
              <div
                className="ms-TooltipHost host_a3f87333"
                onBlurCapture={[Function]}
                onFocusCapture={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              />
            </span>
            <i
              aria-hidden={true}
              className="ms-Breadcrumb-chevron chevron-50"
              data-icon-name="ChevronRight"
              role="presentation"
            />
          </li>
          <li
            className="ms-Breadcrumb-listItem listItem-43"
          >
            <span
              className="ms-Breadcrumb-item item-48"
            >
              <div
                className="ms-TooltipHost host_a3f87333"
                onBlurCapture={[Function]}
                onFocusCapture={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              />
            </span>
          </li>
        </ol>
      </div>
    </div>
  </div>
</div>
`;
